raft/raft.hh

```diff
@@ -8,15 +8,12 @@
 #pragma once
 
 #include "utils/assert.hh"
-#include <concepts>
```

Comment:
[L818](https://github.com/scylladb/scylladb/blob/4346a974197f5b448b907f09402eb8eeb652b6d1/raft/raft.hh#L818C11-L818C28) uses `std::derived_from` which is declared in `<concepts>`. It is guarded by an `#if` directive.
Either:
Keep the include if this `#if` can appear.  or
Remove the code under the `#if` if we don't need it anymore.

Reply:
we don't support fmt < v10 anymore. do you think the cleanup is a prerequisite of this change?

Reply:
I think that leaving unsupported code is a bad idea.
Then yes, please delete the unsupported code (and remove the include as well.)

Reply:
fair enough.  i don't think the fmt v10 cleanup should block this change. but here it is: #21847

Reply:
Thanks for #21847!